### PR TITLE
LINK-1880 | Automatically Refund Signup Payments

### PR DESCRIPTION
**Pull request by harriris-vincit**
### Description
Automatically do a Talpa refund request when cancelling a signup or a signup group that has an unrefunded payment.
### Closes
[LINK-1880](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1880)

[LINK-1880]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1880?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

File: registrations/models.py [hunk 1]
```diff
@@ -705,6 +708,43 @@ class SignUpGroup(
         {"name": "contact_person"},
     )
 
+    @transaction.atomic
+    def delete(self, *args, **kwargs):
+        contact_person = getattr(self, "contact_person", None)
+        is_attending = True if self.attending_signups else False
```
Comment (charn) [inline registrations/models.py:714]: You can also do:
```suggestion
        is_attending = bool(self.attending_signups)
```

File: web_store/tests/payment/test_web_store_payment_api_client.py [hunk 2]
```diff
@@ -76,3 +121,34 @@ def test_get_payment_exception(status_code):
...
+        status.HTTP_403_FORBIDDEN,
+        status.HTTP_404_NOT_FOUND,
+        status.HTTP_500_INTERNAL_SERVER_ERROR,
+    ],
+)
+def test_create_refund_exception(status_code):
+    client = WebStorePaymentAPIClient()
+    mocked_response = get_mock_response(status_code=status_code)
+
+    with patch("requests.get") as mocked_request, pytest.raises(RequestException):
```
Comment (charn) [inline web_store/tests/payment/test_web_store_payment_api_client.py:152]: Just a note that the project also has `requests-mock` package which can be used for mocking requests.

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

File: registrations/models.py (hunk 1 above)
Comment (harriris-vincit) [reply registrations/models.py]: Yes indeed, this was the solution I wanted yesterday afternoon. Now changed as suggested.

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

File: web_store/tests/payment/test_web_store_payment_api_client.py (hunk 2 above)
Comment (harriris-vincit) [reply web_store/tests/payment/test_web_store_payment_api_client.py:152]: I created a more general refactoring ticket from this note to our backlog: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1957.

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr863.api.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**